backend/persistent/datastores/DataWorld.js

```diff
@@ -0,0 +1,118 @@
+import fetch from 'node-fetch';
+import url from 'url';
+
+import Logger from '../../logger';
+
+function parseUrl(datasetUrl) {
+  const pathnameArray = url.parse(datasetUrl).pathname.split('/');
+  return {
+    owner: pathnameArray[1],
+    id: pathnameArray[2]
+  };
+}
+
+export function connect(connection) {
+  const { owner, id } = parseUrl(connection.url);
+  return new Promise((resolve, reject) => {
```

Comment:
No need to create a new Promise, just `return fetch(...` and replace `reject(json)` by `throw new Error(JSON.stringify(json))`.